### PR TITLE
refactor: add method params to storage power actor

### PR DESCRIPTION
**Pull request by frrist**
mechanical change, all exported methods accept a params struct instead of individual arguments. Added code gen as well.

### CODE REVIEW

File: actors/builtin/storage_power/storage_power_actor.go [hunk 1]
```diff
@@ -21,9 +21,11 @@ import (
 )
 
 type Runtime = vmr.Runtime
+
 var Assert = autil.Assert
 
 type ConsensusFaultType int
+
```
Comment (frrist) [inline actors/builtin/storage_power/storage_power_actor.go]: gofmt

File: actors/builtin/storage_power/storage_power_actor.go [hunk 2]
```diff
@@ -37,14 +39,41 @@ type StoragePowerActor struct{}
 // Actor methods
 ////////////////////////////////////////////////////////////////////////////////
 
-func (a *StoragePowerActor) AddBalance(rt Runtime, minerAddr addr.Address) *adt.EmptyValue {
-	builtin.RT_MinerEntry_ValidateCaller_DetermineFundsLocation(rt, minerAddr, builtin.MinerEntrySpec_MinerOnly)
+func (a *StoragePowerActor) Constructor(rt Runtime) *adt.EmptyValue {
```
Comment (frrist) [inline actors/builtin/storage_power/storage_power_actor.go]: moved method to top of file, no logic change

File: actors/builtin/storage_power/storage_power_actor.go [hunk 3]
```diff
@@ -159,69 +203,90 @@ func (a *StoragePowerActor) DeleteMiner(rt Runtime, minerAddr addr.Address) *adt
...
 	}
-	if err := a.addPowerForSector(rt, rt.ImmediateCaller(), newWeight); err != nil {
+	if err := a.addPowerForSector(rt, rt.ImmediateCaller(), params.newWeight); err != nil {
 		rt.Abort(exitcode.ErrIllegalState, "Failed to add power for sector: %v", err)
 	}
 	return &adt.EmptyValue{}
 }
 
-func (a *StoragePowerActor) OnMinerSurprisePoStSuccess(rt Runtime) *adt.EmptyValue {
+func (a *StoragePowerActor) OnMinerSurprisePoStSuccess(rt Runtime, _ *adt.EmptyValue) *adt.EmptyValue {
```
Comment (frrist) [inline actors/builtin/storage_power/storage_power_actor.go]: To keep method signatures (`func (Runtime, CBORMarshal)`) the same I've introduced this parameter, any objection?

File: actors/builtin/storage_power/storage_power_actor.go [hunk 4]
```diff
@@ -37,14 +39,41 @@ type StoragePowerActor struct{}
 // Actor methods
 ////////////////////////////////////////////////////////////////////////////////
 
-func (a *StoragePowerActor) AddBalance(rt Runtime, minerAddr addr.Address) *adt.EmptyValue {
-	builtin.RT_MinerEntry_ValidateCaller_DetermineFundsLocation(rt, minerAddr, builtin.MinerEntrySpec_MinerOnly)
+func (a *StoragePowerActor) Constructor(rt Runtime, _ *adt.EmptyValue) *adt.EmptyValue {
```
Comment (frrist) [inline actors/builtin/storage_power/storage_power_actor.go]: To keep method signatures (func (Runtime, CBORMarshal)) the same I've introduced this parameter, any objection?

File: actors/builtin/storage_power/storage_power_actor.go [hunk 5]
```diff
@@ -159,69 +203,89 @@ func (a *StoragePowerActor) DeleteMiner(rt Runtime, minerAddr addr.Address) *adt
...
 	}
-	if err := a.addPowerForSector(rt, rt.ImmediateCaller(), newWeight); err != nil {
+	if err := a.addPowerForSector(rt, rt.ImmediateCaller(), params.newWeight); err != nil {
 		rt.Abort(exitcode.ErrIllegalState, "Failed to add power for sector: %v", err)
 	}
 	return &adt.EmptyValue{}
 }
 
-func (a *StoragePowerActor) OnMinerSurprisePoStSuccess(rt Runtime) *adt.EmptyValue {
+func (a *StoragePowerActor) OnMinerSurprisePoStSuccess(rt Runtime, _ *adt.EmptyValue) *adt.EmptyValue {
```
Comment (frrist) [inline actors/builtin/storage_power/storage_power_actor.go]: and here

File: actors/builtin/storage_power/storage_power_actor.go [hunk 6]
```diff
@@ -342,14 +423,14 @@ func (a *StoragePowerActor) ReportConsensusFault(rt Runtime, blockHeader1, block
...
-	if err := a.deleteMinerActor(rt, slashee); err != nil {
+	if err := a.deleteMinerActor(rt, params.slashee); err != nil {
 		rt.Abort(exitcode.ErrIllegalState, "Failed to delete miner actor: %v", err)
 	}
 	return &adt.EmptyValue{}
 }
 
 // Called by Cron.
-func (a *StoragePowerActor) OnEpochTickEnd(rt Runtime) *adt.EmptyValue {
+func (a *StoragePowerActor) OnEpochTickEnd(rt Runtime, _ *adt.EmptyValue) *adt.EmptyValue {
```
Comment (frrist) [inline actors/builtin/storage_power/storage_power_actor.go]: and here

File: actors/builtin/storage_power/storage_power_actor.go (hunk 4 above)
Comment (anorth) [reply actors/builtin/storage_power/storage_power_actor.go]: Beware conflict with #68

**Pull request merged**